### ISSUE TITLE
Match names restricted to a facet's values (from 533)

### ISSUE THREAD

**Issue by gigamorph** [project-lux/lux-marklogic#6]
For some of the facets, the user should be allowed to enter part of a facet value's name and be presented with matching names.

At present, the backend does not include names as part of a facet response.  And Kam is concerned this would be too much to do client side --it presently only retrieves the subset of names displayed --the number of values for some facets for some searches could really bog the frontend down.

An implementation choice is to extend the auto complete endpoint to accept search criteria and restrict matches to the provided text and search criteria.  To our benefit, auto complete's implementation (#55) already uses cts.search and related list's implementation (#479) already refactored the part of the search library that generates queries from a search string.

Which facets are in scope of this ticket?  Auto complete's current implementation is well suited to be immediately extended for _non_-semantic facets restricted by _non_-semantic search criteria.  Thus, that's what I would limit the scope of this ticket to.

---

Beta 1's means to resolve semantic and non-semantic search criteria via CTS Search partially invalidates the following.  That said, when it comes to facets, #637 had us switch to cts.fieldValues which should continue to encourage us to limit this ticket's scope to non-semantic facets --those backed by a field range index.

~A separate ticket could be created for semantic requirements; such a ticket could be implemented after:~

* ~We restore store for semantic search criteria: upgrade to ML 11 and #504.~
* ~We introduce the ability to restrict facet values by semantic search criteria: #420.~
* ~We have implemented the semantic search facets.  Related lists aside, we have not implemented a semantic facet.  Tickets requesting specific ones include #222, #247, #248, and #469.~

---

Auto complete's performance has not been proven.  Response time can vary by text + index combination, as well as concurrent demand.  However, typically cts.search's performance improves when more criteria is specified.

For current auto complete capabilities, please see the [auto complete endpoint documentation](https://git.yale.edu/lux-its/marklogic/blob/main/docs/lux-backend-api-usage.md#auto-complete).  The endpoint consumer may presently specify whether the match should be against primary names or primary and alternative names.

#55 enabled auto complete for the following fields:

- agentName
- agentPrimaryName
- anyName
- placeName
- placePrimaryName
- conceptName
- conceptPrimaryName

If the in-scope facets are associated to additional fields, we'll need to ensure there is also a field range index for it, likely be changing the "Force Range Index?" column value to 'Y' in the [database index configuration spreadsheet](https://docs.google.com/spreadsheets/d/1lwn2tzRNRHtLTPMNWvOsy5fYiIfY4wkthhvgctQ8Hpg/edit#gid=997981330) then regenerating the database index configuration as documented in [Database Index Configuration](https://git.yale.edu/lux-its/marklogic/blob/main/conf/README.md#database-index-configuration).  If this is determined necessarily, please add the "re-indexing required" label. 

cc: @pc754, @sk969

**Comment by brent-hartwig** [comment]
@roamye, should this be marked as blocking https://github.com/project-lux/lux-frontend/issues/1 versus being blocked?

**Comment by roamye** [comment]
Based on this comment (Ticket from old repo) | @prowns : 
https://git.yale.edu/lux-its/marklogic/issues/533#issuecomment-23143, this issue was listed as blocked by https://github.com/project-lux/lux-frontend/issues/1 as it needs updated requirements. 

If that should no longer be the case then I can move it as you suggested. @brent-hartwig

**Comment by brent-hartwig** [comment]
@roamye, ok, yes, if https://github.com/project-lux/lux-frontend/issues/1's requirements must be figured out before this ticket can be prioritized, I agree this ticket is blocked.  Thanks for clarifying.
